### ISSUE TITLE
Define architecture in Package.json and use it to build. 

### ISSUE THREAD

**Issue by seivan** [dprint/dprint#555]
Not exactly sure how, but I know it's possible because ESBuild does that.
If you could define the [arch/CPU](https://github.com/dprint/dprint/blob/main/deployment/npm/package.json) matrix supported, and use those values for the install script, then you could let fetch each prebuilt binary per arch/cpu matrix. 

We're using Yarn Berry, and that means we're bundling our dependencies (including the formatter) for a zero-install mode. 
So when i install dependencies, if a dependency have multiple arch/cpu defined and supported, it will fetch them based on my config.

For instance, I've defined 
```ts
supportedArchitectures:
  cpu:
    - arm64
    - x64
  os:
    - darwin
    - linux
    ```
    
So it will make sure that whatever build/install script that runs after fetching will use those values form us, to fetch/compile each version. 

Even though I might be `Darwin-arm64` the `CI` will probably be `linux-x64`. But the CI doesn't actually install the dependencies, it just checks the lock-file compared to a new install (and fails if it doesn't match).

**Comment by dsherret** [comment]
@seivan the dprint npm package in 0.32.1 will now auto-initialize on first run in case the post install script was never run. Maybe this fixes your scenario?

**Comment by dsherret** [comment]
This should be fixed now and the upcomming fix to #686 will make it work with yarn berry.
